feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,755 @@
+	sort.Slice(ports, func(i, j int) bool {
+		idi, idj := ports[i].ID(), ports[j].ID()
+		li, lj := len(idi), len(idj)
+		if li == lj {
+			return idi < idj
+		}
+		return li < lj // "port2" < "port10"
+	})
+	return ports
+}
+
+func contains(list []string, target string) bool {
+	for _, item := range list {
+		if item == target {
+			return true
+		}
+	}
+	return false
+}
+
+func removeElement(list []string, element string) []string {
+	for i := 0; i < len(list); i++ {
+		if list[i] == element {
+			list = append(list[:i], list[i+1:]...)
+			i-- // Adjust index to account for removed element
+		}
+	}
+	return list
+}
+
+func TestCheckForCoreFiles(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	timestamp := uint64(time.Now().UTC().Unix())
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	coreFileCheck(t, dut, gnoiClient, timestamp, true)
+}
+
+func TestComponentStatus(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
+	lineCards := components.FindComponentsByType(t, dut, lineCardType)
+	fabricCards := components.FindComponentsByType(t, dut, fabricCardType)
+	checkComponents := append(controllerCards, lineCards...)
+	checkComponents = append(checkComponents, fabricCards...)
+	// check oper-status of the components is Active.
+	for _, component := range checkComponents {
+		val, present := gnmi.Lookup(t, dut, gnmi.OC().Component(component).OperStatus().State()).Val()
+		if !present {
+			t.Errorf("ERROR: Get component %s oper-status failed", component)
+		}
+		if present && val != oc.PlatformTypes_COMPONENT_OPER_STATUS_ACTIVE {
+			t.Errorf("ERROR: Get component %s oper-status failed", component)
+		}
+	}
+
+	if deviations.ConsistentComponentNamesUnsupported(dut) {
+		t.Skipf("Skipping test due to deviation consistent_component_names_unsupported")
+	}
+
+	// use gNOI to check components health status is not "unhealthy".
+	gnoiClient := dut.RawAPIs().GNOI().New(t)
+	for _, component := range checkComponents {
+		componentName := map[string]string{"name": component}
+		req := &hpb.GetRequest{
+			Path: &tpb.Path{
+				Elem: []*tpb.PathElem{
+					{Name: "components"},
+					{
+						Name: "component",
+						Key:  componentName,
+					},
+				},
+			},
+		}
+		validResponse, err := gnoiClient.Healthz().Get(context.Background(), req)
+		if err != nil {
+			t.Errorf("Error: %v", err)
+			continue
+		}
+		if validResponse.GetComponent().Status == hpb.Status_STATUS_UNHEALTHY {
+			t.Errorf("Found unhealthy component: %s", component)
+		} else {
+			t.Logf("Response: %+v", validResponse)
+		}
+	}
+}
+
+func TestControllerCardsNoHighCPUSpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	if deviations.ControllerCardCPUUtilizationUnsupported(dut) {
+		t.Skipf("Skipping test due to deviation linecard_cpu_ultilization_unsupported")
+	}
+
+	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
+	cardList := controllerCards
```

Comment:
No need to declare a new variable cardList. The "controllerCards" variable can just be reused below instead.

Reply:
Addressed with the latest commit. Thanks!

Reply:
Thanks for your comments. Removed "cardList" variable and reuse "controllerCards":
```go
	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
	cpuCards := components.FindComponentsByType(t, dut, cpuType)
	for _, cpu := range cpuCards {
```